latch/resources/workflow.py

```diff
@@ -1,95 +1,94 @@
-
-            if len(in_meta_not_in_wf) > 0 or len(not_in_meta_in_wf) > 0:
-                error_str = (
-                    "Inconsistency detected between parameters in your `LatchMetadata`"
-                    " object and parameters in your workflow signature.\n\n"
-                )
-
-                if len(in_meta_not_in_wf) > 0:
-                    error_str += (
-                        "The following parameters appear in your `LatchMetadata` object"
-                        " but not in your workflow signature:\n\n"
-                    )
-                    for meta_param in in_meta_not_in_wf:
-                        error_str += f"    \x1b[1m{meta_param}\x1b[22m\n"
-                    error_str += "\n"
-
-                if len(not_in_meta_in_wf) > 0:
-                    error_str += (
-                        "The following parameters appear in your workflow signature but"
-                        " not in your `LatchMetadata` object:\n\n"
-                    )
-                    for meta_param in not_in_meta_in_wf:
-                        error_str += f"    \x1b[1m{meta_param}\x1b[22m\n"
-                    error_str += "\n"
-
-                error_str += (
-                    "Please resolve these inconsistencies and ensure that your"
-                    " `LatchMetadata` object and workflow signature have the same"
-                    " parameters."
-                )
-
-                raise ValueError(error_str)
-
-            for name, meta_param in metadata.parameters.items():
-                if meta_param.samplesheet is not True:
-                    continue
-
-                annotation = wf_params[name].annotation
-
-                origin = get_origin(annotation)
-                args = get_args(annotation)
-                valid = (
-                    origin is not None
-                    and issubclass(origin, list)
-                    and is_dataclass(args[0])
+        f = metadata
+        signature = inspect.signature(f)
+        wf_params = signature.parameters
+        metadata = LatchMetadata(f.__name__, LatchAuthor())
+        metadata.parameters = _generate_params(wf_params)
+
+        _inject_metadata(f, metadata)
+        return _workflow(f)
+
+    def decorator(f: Callable):
+        signature = inspect.signature(f)
+        wf_params = signature.parameters
+
+        nonlocal metadata
+        if metadata is None:
+            metadata = LatchMetadata(f.__name__, LatchAuthor())
+            metadata.parameters = _generate_params(wf_params)
+
+        for wf_param in wf_params:
+            if wf_param not in metadata.parameters:
+                metadata.parameters[wf_param] = LatchParameter(display_name=wf_param)
+
+        in_meta_not_in_wf = []
+        for meta_param in metadata.parameters:
+            if meta_param not in wf_params:
+                in_meta_not_in_wf.append(meta_param)
+        if len(in_meta_not_in_wf) > 0:
```

Comment:
```suggestion
            if meta_param not in wf_params:
                in_meta_not_in_wf.append(meta_param)
                
        if len(in_meta_not_in_wf) > 0:
```

personal style preference is to have a newline between any two blocks at the same indentation level